x/evm/module.go

```diff
@@ -192,5 +194,23 @@ func (am AppModule) EndBlock(ctx context.Context) error {
 			}
 		}()
 	}
+
+	if sdkCtx.BlockHeight()%updateXChainsReferencesPeriod == 0 {
+		func() {
+			cis, err := am.keeper.GetAllChainInfos(sdkCtx)
+			if err != nil {
+				am.keeper.Logger(sdkCtx).Error("error while trying to get all chain infos to refresh reference blocks", "error", err)
+				return
+			}
+
+			for _, ci := range cis {
+				err := am.keeper.UpdateReferenceBlockForChain(sdkCtx, ci.GetChainReferenceID())
+				if err != nil {
+					am.keeper.Logger(sdkCtx).Error("error while adding request to refresh reference block for chain", "error", err, "chain-reference-id", ci.GetChainReferenceID())
+				}
+			}
+		}()
```

Comment:
Why not extract this into testable function that returns an error. That way you can just call it here, and log once.

Also, use `liblog` for logging.